realm/realm-library/src/main/java/io/realm/internal/async/QueryUpdateTask.java

```diff
@@ -100,8 +101,17 @@ public void run() {
                 handler.obtainMessage(message, result).sendToTarget();
             }
 
-        } catch (Exception e) {
+        } catch (BadVersionException e) {
+            // In some rare race conditions, this can happen. In that case, just ignore the error.
```

Comment:
Good catch. Yes it should. Although with 1.2, the handover object should now prevent the version from being invalidated, so I think we can actually remove all our code that handles the BadVersionException, but I would do that in another PR as it needs testing.